### PR TITLE
Quote items qty validation tests on default and not default stocks

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
1. #520 
Integrational tests for check whether a customer can buy or put into Shopping Cart needed an amount of product

### CODE REVIEW

File: app/code/Magento/InventorySales/Test/Integration/SalesQuoteItem/AddSalesQuoteItemOnDefaultStockTest.php [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    /**
+     * @param string $sku
+     * @return ProductInterface
+     * @throws NoSuchEntityException
+     */
+    private function getProductBySku(string $sku): ProductInterface
+    {
+        $product = $this->productRepository->get($sku);
+        $product->setIsSalable(true);
```
Comment (naydav) [inline app/code/Magento/InventorySales/Test/Integration/SalesQuoteItem/AddSalesQuoteItemOnDefaultStockTest.php]: You don't need to set is_sallable flag forcible
If product has wrong status it is a bug

Comment (VitaliyBoyko) [reply app/code/Magento/InventorySales/Test/Integration/SalesQuoteItem/AddSalesQuoteItemOnDefaultStockTest.php]: Done. Thanks.

File: app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessRegisterProductsSalePlugin.php [hunk 2]
```diff
@@ -81,9 +90,16 @@ public function aroundRegisterProductsSale(StockManagement $subject, callable $p
 
         $reservations = [];
         foreach ($productSkus as $productId => $sku) {
+            $qty = (float)$items[$productId];
+            $isSalable = $this->isProductSalableForRequestedQty->execute($sku, $stockId, $qty)->isSalable();
+            if (!$isSalable) {
+                throw new LocalizedException(
+                    __('Not all of your products are available in the requested quantity.')
+                );
+            }
```
Comment (maghamed) [inline app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessRegisterProductsSalePlugin.php]: we need to make sure that  we can fulfill the order for all items and just after that place all the reservations.

So, you will have two loops (first - with isProductSalableForRequestedQty call  and all the preparation, second - create all the reservations )

File: app/code/Magento/InventorySales/Test/Integration/Order/PlaceOrderOnDefaultSourceTest.php [hunk 3]
```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * @author Atwix Team
+ * @copyright Copyright (c) 2018 Atwix (https://www.atwix.com/)
```
Comment (maghamed) [inline app/code/Magento/InventorySales/Test/Integration/Order/PlaceOrderOnDefaultSourceTest.php]: could you please fix doc type according to Magento requirements

File: app/code/Magento/InventorySales/Test/Integration/Order/PlaceOrderOnNotDefaultSourceTest.php [hunk 4]
```diff
@@ -0,0 +1,276 @@
+<?php
+/**
+ * @author Atwix Team
+ * @copyright Copyright (c) 2018 Atwix (https://www.atwix.com/)
+ */
```
Comment (maghamed) [inline app/code/Magento/InventorySales/Test/Integration/Order/PlaceOrderOnNotDefaultSourceTest.php]: please fix the header

**Review by maghamed — CHANGES_REQUESTED** [review]

**Pull request merged**